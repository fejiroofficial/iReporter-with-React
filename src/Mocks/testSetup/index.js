/* eslint-disable import/no-extraneous-dependencies */
import '@babel/polyfill';
import React from 'react';
import { Provider } from 'react-redux';
import { mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import thunk from 'redux-thunk';
import configMockStore from 'redux-mock-store';
import { BrowserRouter } from 'react-router-dom';


export const mockState = {
  user: {
    token: '',
    isLoggedIn: false,
    loginStatus: false,
    message: '',
    isLoading: false,
    profile: {
      firstname: 'John',
      username: 'johnny',
      image: 'image.jpg',
      errorData: [],
    },
    resetPassword: {
      loading: false,
      message: '',
    }
  },
  bookmark: {
    loading: false,
    articles: [],
    message: '',
  },
  article: {
    newComments: {
      body: 'Hello world',
      id: 1,
      authorName: 'John',
      authorImage: 'image.jpg',
      date: 'July 4th 2018',
    },
    postingComment: false,
    commentMessage: 'Done',
    gettingComments: false,
    oldComments: [],
    hasMoreComments: false,
    commentPage: {
      current: 1,
      currentCount: 1,
      totalCount: 1,
    },
  },
  createArticle: {
    loading: false,
    message: false,
  },
  articlesReducer: {
    articles: []
  },
  articlesCategoryReducer: []
};


export const createMockStore = () => {
  const mockStore = configMockStore([thunk]);
  configure({ adapter: new Adapter() });
  return mockStore;
};

const setup = (component, initailState = mockState) => {
  const mockStore = configMockStore([thunk]);
  configure({ adapter: new Adapter() });
  const connectedWrapper = mount(
    <Provider store={mockStore(initailState)}>
      <BrowserRouter>
        {component}
      </BrowserRouter>
    </Provider>
  );
  return connectedWrapper;
};

export default setup;
