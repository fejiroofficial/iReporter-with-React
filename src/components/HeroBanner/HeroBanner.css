.HeroBanner {
    height: 181px;
    width: inherit;
    background-image: url('../.././assets/img/hero.jpg');
    background-size: cover;
    background-position: -27%;
    background-repeat: no-repeat;
    padding-top: 100px;
}

.HeroBanner h1 {
    font-size: 16px;
    line-height: 37px;
    text-align: center;
    color: #503f3f;
    display: none;
}

.HeroBanner button {
  display: none;
}

.SmallScreenH1 {
  font-size: 16px;
  line-height: 37px;
  text-align: center;
  color: #503f3f;
}

.SmallScreenHeaderBtnBox {
  height: 50px;
  width: inherit;
  text-align: center;
}

.SmallScreenHeaderBtnBox button {
  height: 40px;
  width: 201px;
  border-width: thin;
  border-radius: 20px;
  font-size: 15px;
  text-align: center;
  color: white;
  background-color: lightseagreen;
  cursor: pointer;
  outline: none;
}

.Seperator {
  height: 1px;
  width: inherit;
  background-color: lightgrey;
  margin-top: 11px;
}

@media (min-width: 768px) {
    .HeroBanner {
    height: 423px;
    }
}


@media (min-width: 1024px) {
    .HeroBanner {
        height: 603px;
        padding-top: 51px;
        padding-left: 63px;
        background-position: 14%;
        background-position-y: top;
        background-size: 143%;
    }
    .HeroBanner h1 {
      font-size: 68px;
      width: 561px;
      line-height: 71px;
      text-align: center;
      color: #706363;
      display: block;
    }
    .HeroBanner button {
      display: block;
      height: 49px;
      width: 344px;
      font-weight: bold;
      font-size: 20px;
      border-width: thin;
      border-radius: 20px;
      margin-left: 98px;
      text-align: center;
      color: white;
      background-color: lightseagreen;
      cursor: pointer;
    }
    .HeroBanner button:hover {
      background-color: #1f8983;
    }
    .SmallScreenH1 {
      display: none;
    }
    .SmallScreenHeaderBtnBox button {
        display: none;
    }
    .Seperator {
      display: none;
    }
    .SmallScreenHeaderBtnBox {
      display: none;
    }
}

@media (min-width: 1359px) {
  .HeroBanner {
    padding-left: 180px;
  }
}