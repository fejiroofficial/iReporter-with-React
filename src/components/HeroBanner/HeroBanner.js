import React from 'react';
import { Link } from 'react-router-dom';
import Classes from './HeroBanner.css';
import Aux from '../../hoc/Aux';

const heroBanner = props => (
  <Aux>
    <div className={Classes.HeroBanner}>
      <h1>Are you Ready To Fight Corruption?</h1>
      <Link to="/signup">
        <button>Sign Up For Free</button>
      </Link>
    </div>
    <h1 className={Classes.SmallScreenH1}>
      Are you Ready To Fight Corruption?
    </h1>
    <div className={Classes.SmallScreenHeaderBtnBox}>
      <Link to="/signup">
        <button>Sign Up For Free</button>
      </Link>
    </div>
    <div className={Classes.Seperator}></div>
  </Aux>
);

export default heroBanner;
