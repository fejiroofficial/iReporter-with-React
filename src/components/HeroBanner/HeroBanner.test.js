import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import { createMockStore } from '../../Mocks/testSetup';
import HeroBanner from './HeroBanner';

createMockStore();

describe('<HeroBanner>', () => {
  it('should render without crashing', () => {
    const container = (
      <HeroBanner>
        <p>Hello</p>
      </HeroBanner>
    );
    const wrapper = shallow(container);
    expect(wrapper).toMatchSnapshot();
  });
});
