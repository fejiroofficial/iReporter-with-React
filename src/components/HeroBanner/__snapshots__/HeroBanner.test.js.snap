// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeroBanner> should render without crashing 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
